src/lib/swapper/swappers/ThorchainSwapper/endpoints.ts

```diff
@@ -19,14 +22,60 @@ export const thorchainApi: Swapper2Api = {
   getTradeQuote: async (
     input: GetTradeQuoteInput,
     rates: Rates,
+    runeAssetUsdRate: string,
   ): Promise<Result<TradeQuote2, SwapErrorRight>> => {
-    const tradeQuoteResult = await getThorTradeQuote(input, rates)
+    const { receiveAddress, affiliateBps } = input
 
-    return tradeQuoteResult.map(tradeQuote => {
-      const { receiveAddress, affiliateBps } = input
+    const mapTradeQuoteToTradeQuote2 = (
+      quote: Result<TradeQuote, SwapErrorRight>,
+      receiveAddress: string,
+      affiliateBps: string | undefined,
+      isDonationAmountBelowMinimum?: boolean,
+    ): Result<TradeQuote2, SwapErrorRight> => {
       const id = uuid()
+      return quote.map<TradeQuote2>(quote => ({
+        id,
+        receiveAddress,
+        affiliateBps: isDonationAmountBelowMinimum ? undefined : affiliateBps,
+        ...quote,
+      }))
+    }
+
+    return await getThorTradeQuote(input, rates).then(async firstQuote => {
+      if (firstQuote.isOk()) {
+        const successfulQuote = firstQuote.unwrap()
+        const firstHop = successfulQuote.steps[0]
+        const buyAmountBeforeFeesCryptoPrecision = fromBaseUnit(
+          firstHop.buyAmountBeforeFeesCryptoBaseUnit,
+          firstHop.buyAsset.precision,
+        )
+        const buyAmountUsd = bnOrZero(buyAmountBeforeFeesCryptoPrecision).times(
+          rates.buyAssetUsdRate,
+        )
+        const donationAmountUsd = buyAmountUsd.times(affiliateBps).div(10000)
+        const isDonationAmountBelowMinimum =
+          bnOrZero(affiliateBps).gt(0) &&
+          bnOrZero(donationAmountUsd)
+            .div(runeAssetUsdRate)
+            .lte(RUNE_OUTBOUND_TRANSACTION_FEE_CRYPTO_HUMAN)
 
-      return { id, receiveAddress, affiliateBps, ...tradeQuote }
+        const quoteToUse = isDonationAmountBelowMinimum
+          ? /*
+          If the donation amount is below the minimum,
+          we need to fetch a new quote with no affiliate fee
+        */
+            await getThorTradeQuote({ ...input, affiliateBps: '0' }, rates)
+          : firstQuote
+
+        return mapTradeQuoteToTradeQuote2(
+          quoteToUse,
+          receiveAddress,
+          affiliateBps,
+          isDonationAmountBelowMinimum,
+        )
+      } else {
+        return mapTradeQuoteToTradeQuote2(firstQuote, receiveAddress, affiliateBps)
```

Comment:
Prefer to move this as an early return, possibly with a comment explaining we're passing the errorred quote through, bit confusing IMO

Reply:
Used an early return and added a comment in https://github.com/shapeshift/web/pull/4902/commits/d7641e10ca68c0fa631a463acb55c5c0953be8c8 to make things a little clearer.

Reply:
Much clearer now, ty ser!